index.bs

```diff
@@ -795,6 +741,528 @@ the following list:
+</div>
+
+
+# Feedback to WebSocketStream from the protocol # {#feedback-to-websocket-stream-from-the-protocol}
+
+When [=the WebSocket connection is established=], the user agent must [=queue a task=] to run these
+steps:
+
+<div algorithm="handle WebSocketStream connection establishment">
+
+ 1. Change the [=WebSocketStream/ready state=] to {{WebSocket/OPEN}} (1).
+ 1. Set |this|'s [=WebSocketStream/was ever connected flag=].
+ 1. Let |extensions| be the [=extensions in use=].
+ 1. Let |protocol| be the [=subprotocol in use=].
+ 1. Let |pullAlgorithm| be an action that [=pulls bytes=] from the WebSocket.
+ 1. Let |cancelAlgorithm| be an action that [=cancels=] the WebSocketStream with |this| and |reason|, given
+    |reason|.
+ 1. Let |readable| be a [=new=] {{ReadableStream}}.
+ 1. [=ReadableStream/Set up=] |readable| with |pullAlgorithm| and |cancelAlgorithm|.
+ 1. Let |writeAlgorithm| be an action that [=writes=] |chunk| to |this|, given |chunk|.
+ 1. Let |closeAlgorithm| be an action that [=closes=] |this|.
+ 1. Let |abortAlgorithm| be an action that [=aborts=] |this| with |reason|, given |reason|.
+ 1. Let |writable| be a [=new=] {{WritableStream}}.
+ 1. [=WritableStream/Set up=] |writable| with |writeAlgorithm|, |closeAlgorithm|, and
+    |abortAlgorithm|.
+ 1. Set |this|'s [=WebSocketStream/readable stream=] to |readable|.
+ 1. Set |this|'s [=WebSocketStream/writable stream=] to |writable|.
+ 1. Let |openInfo| be a [=new=] {{WebSocketOpenInfo}}.
+ 1. Set |openInfo|["{{WebSocketOpenInfo/extensions}}"] to |extensions|.
+ 1. Set |openInfo|["{{WebSocketOpenInfo/protocol}}"] to |protocol|.
+ 1. Set |openInfo|["{{WebSocketOpenInfo/readable}}"] to |readable|.
+ 1. Set |openInfo|["{{WebSocketOpenInfo/writable}}"] to |writable|.
+ 1. [=Resolve=] |this|'s [=WebSocketStream/opened promise=] with |openInfo|.
+
+</div>
+
+<hr>
+
+<div algorithm="handle a WebSocketStream message">
+
+When [=a WebSocket message has been received=] for |this| with type |type| and data |data|, the user
+agent must [=queue a task=] to follow these steps: [[!WSP]]
+
+ 1. If [=WebSocket/ready state=] is not {{WebSocket/OPEN}} (1), then return.
+ 1. Let |chunk| be determined by switching on |type|:
+
+  <dl class="switch">
+   : |type| indicates that the data is Text
+   :: a new {{DOMString}} containing |data|
+
+   : |type| indicates that the data is Binary
+   :: a new {{Uint8Array}} object, created in the [=relevant Realm=] of the {{WebSocketStream}}
+      object, whose contents are |data|
+  </dl>
+
+ 1. [=ReadableStream/Enqueue=] |chunk| into |this|'s [=WebSocketStream/readable stream=].
+ 1. Apply backpressure to the WebSocket.
+
+ Note: Applying backpressure <span class=allow-2119>should</span> result in no new WebSocket
+ messages being handled until the backpressure is released. After an implementation-defined amount
+ of data is buffered, the client will stop reading from the TCP/IP socket resulting in network-layer
```

Comment:
If this is running over H3, we shouldn't assume TCP
```suggestion
 of data is buffered, the client will stop reading from the underlying network connection resulting in network-layer
```

Reply:
Good point.